docs/developer/extending.rst

```diff
@@ -0,0 +1,328 @@
+        # 'myipam',   <-- replace without your app-name here
+        # admin
+        "admin_auto_filters",
+        "django.contrib.admin",
+        # rest framework
+        "rest_framework",
+        # Other dependencies
+        "reversion",
+    ]
+
+For more information about how to work with django projects and django
+apps, please refer to the `django documentation
+<https://docs.djangoproject.com/en/dev/intro/tutorial01/>`_.
+
+2. Install ``openwisp-ipam``
+----------------------------
+
+Install (and add to the requirement of your project) openwisp-ipam:
+
+.. code-block::
+
+    pip install openwisp-ipam
+
+3. Add ``EXTENDED_APPS``
+------------------------
+
+Add the following to your ``settings.py``:
+
+.. code-block:: python
+
+    EXTENDED_APPS = ("openwisp_ipam",)
+
+4. Add ``openwisp_utils.staticfiles.DependencyFinder``
+------------------------------------------------------
+
+Add ``openwisp_utils.staticfiles.DependencyFinder`` to
+``STATICFILES_FINDERS`` in your ``settings.py``:
+
+.. code-block:: python
+
+    STATICFILES_FINDERS = [
+        "django.contrib.staticfiles.finders.FileSystemFinder",
+        "django.contrib.staticfiles.finders.AppDirectoriesFinder",
+        "openwisp_utils.staticfiles.DependencyFinder",
+    ]
+
+5. Add ``openwisp_utils.loaders.DependencyLoader``
+--------------------------------------------------
+
+Add ``openwisp_utils.loaders.DependencyLoader`` to ``TEMPLATES`` in your
+``settings.py``, but ensure it comes before
+``django.template.loaders.app_directories.Loader``:
+
+.. code-block:: python
+
+    TEMPLATES = [
+        {
+            "BACKEND": "django.template.backends.django.DjangoTemplates",
+            "OPTIONS": {
+                "loaders": [
+                    "django.template.loaders.filesystem.Loader",
+                    "openwisp_utils.loaders.DependencyLoader",
+                    "django.template.loaders.app_directories.Loader",
+                ],
+                "context_processors": [
+                    "django.template.context_processors.debug",
+                    "django.template.context_processors.request",
+                    "django.contrib.auth.context_processors.auth",
+                    "django.contrib.messages.context_processors.messages",
+                ],
+            },
+        }
+    ]
+
+6. Inherit the AppConfig Class
+------------------------------
+
+Please refer to the following files in the sample app of the test project:
+
+- `sample_ipam/__init__.py
+  <https://github.com/openwisp/openwisp-ipam/tree/master/tests/openwisp2/sample_ipam/__init__.py>`_.
+- `sample_ipam/apps.py
+  <https://github.com/openwisp/openwisp-ipam/tree/master/tests/openwisp2/sample_ipam/apps.py>`_.
+
+You have to replicate and adapt that code in your project.
+
+For more information regarding the concept of ``AppConfig`` please refer
+to the `"Applications" section in the django documentation
+<https://docs.djangoproject.com/en/dev/ref/applications/>`_.
```

Comment:
please change all the links to point to django 4.2, every time we'll update the recommended django version we'll find & replace across all repos